docker/postgres/entrypoint.sh

```
@@ -1,85 +1,107 @@
+        LEGACY_DIR="$PGDATA/18/docker"
+    elif [ -f "$PGDATA/data/PG_VERSION" ]; then
+        LEGACY_DIR="$PGDATA/data"
     fi
 
-    # Move hidden files (may not exist, so check first)
-    for f in "$LEGACY_DIR"/.[!.]*; do
-        if [ -e "$f" ]; then
-            mv "$f" "$PGDATA/" || { echo "Nimbus: ERROR - Failed to move $f"; exit 1; }
+    if [ -n "$LEGACY_DIR" ] && [ ! -f "$PGDATA/PG_VERSION" ]; then
+        echo "Nimbus: Migrating PostgreSQL data from legacy location ($LEGACY_DIR)..."
+        src_count=$(ls -1A "$LEGACY_DIR" | wc -l)
```

Shellcheck (0.11.0)

[info] Use find instead of ls to better handle non-alphanumeric filenames.

(SC2012)